client/components/forms/clipboard-button/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { Button } from '@automattic/components';
+import classNames from 'classnames';
+import Clipboard from 'clipboard';
+import { useRef, useEffect, forwardRef } from 'react';
+import ReactDom from 'react-dom';
+
+interface ClipboardButtonProps {
+	className?: string;
+	compact?: boolean;
+	disabled?: boolean;
+	primary?: boolean;
+	scary?: boolean;
+	busy?: boolean;
+	borderless?: boolean;
+	plain?: boolean;
+	transparent?: boolean;
+	text: string | null;
+	onCopy?: () => void;
+	onMouseLeave?: () => void;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+const ClipboardButton = forwardRef<
```

Comment:
The only meaningful change in this PR is the addition of React `forwardRef`. We want to pass refs into this component because we'd like to have a tooltip appear when the clipboard button is clicked.

The tooltip requires a reference to the DOM element we want to anchor to. More context here https://github.com/Automattic/wp-calypso/pull/69110

Reply:
Had a separate conversation about this, but it seems like we could just pass the ref as a prop rather than use forwardRef. All things equal, I might prefer that a bit just to avoid the extra syntax complexity involved with forwardRef. 

That said, I don't feel strongly about that, and given that forwardRef is designed for precisely the current purpose and that we use it elsewhere in the codebase, it does seem reasonable to use here.

One question I had was whether this would cause issues with other instances where we're using the component without passing a ref, but if so, then we'd catch that in testing for this PR.

Reply:
> One question I had was whether this would cause issues with other instances where we're using the component without passing a ref, but if so, then we'd catch that in testing for this PR.

This was a really good observation. I assumed `ref` would be some falsey value in this case, but it actually returns an empty object.